addons/website_sale_coupon/controllers/main.py

```diff
@@ -28,6 +49,16 @@ def cart(self, **post):
         order.recompute_coupon_lines()
         return super(WebsiteSale, self).cart(**post)
 
+    @http.route(['/coupon/<string:code>'], type='http', auth='public', website=True, sitemap=False)
+    def activate_coupon(self, code, path='/shop', **kw):
+        request.session['pending_coupon'] = {'promo_code': code}
```

Comment:
when did you handle other things that promo_code?

Reply:
There's also the need to handle the error messages, they must be able to get back to the enduser